vllm/model_executor/models/internlm2.py

```diff
@@ -144,20 +144,23 @@ def __init__(
         )
 
     def split_qkv(self, qkv: torch.Tensor):
-        seq_len = qkv.shape[0]
+        batch_size, seq_len, _ = qkv.shape
```

Comment:
This will break all non-HPU accelerators. Can we make this more generic? E.g. pass `*qkv.shape[:-1]` into reshape, so that 2D tensors will work as well?

Reply:
sure, let me add and test on non-hpu